contracts/PeeranhaUser.sol

```diff
@@ -217,6 +219,37 @@ contract PeeranhaUser is IPeeranhaUser, Initializable, NativeMetaTransaction, Ac
         revokeRole(PROTOCOL_ADMIN_ROLE, userAddr);
     }
 
+    /**
+     * @dev Give bot permission.
+     *
+     * Requirements:
+     *
+     * - Sender must be a bot.
+     * 
+     */
+    function giveBotPermission(address userAddr) public {
```

Comment:
We need to think of changing overall approach here.

There are methods from the base class `AccessControlUpgradeable`:

`_setRoleAdmin(<ADMIN_ROLE>, <ROLE>);`
`grantRole`
`revokeRole`

We only need a single action that we call when add a new role that will execute `_setRoleAdmin`.

CC @Freitag-fri